base/db/R/utils_db.R

```diff
@@ -44,51 +43,98 @@ db.query <- function(query, con=NULL, params=NULL) {
+              "' errorcode=", res$errorNum,
+              " message='", res$errorMsg, "'")
+      )
+    }
   }
-  .db.utils$queries <- .db.utils$queries+1
+  .db.utils$queries <- .db.utils$queries + 1
   invisible(data)
 }
 
-##' Generic function to open a database connection
+##' Open a database connection
 ##'
-##' Create a connection to a database usign the specified parameters. If the paramters contain
-##' driver element it will be used as the database driver, otherwise it will use PostgreSQL.
-##' @name db.open
-##' @title Open database connection
-##' @param params database connection information
-##' @return connection to database
+##' Create a connection to a database using the specified parameters.
+##' The `params` list will be passed as arguments to [DBI::dbConnect()].
+##'
+##' Typical arguments are as follows:
+##' - `driver` -- The name of the database driver. Only `"PostgreSQL"`
+##' and `"Postgres"` are supported. If no driver is specified, default
+##' to `"PostgreSQL"`.
+##' - `user` -- The database username. For local instances of PEcAn,
+##' this is usually `"bety"`.
+##' - `password` -- The database password. For local instances of
+##' PEcAn, this is usually `"bety"`.
+##' - `host` -- The database hostname. For local instances of PEcAn,
+##' this is usually `"localhost"`. Inside the PEcAn Docker stack, this
+##' may be `"postgres"`.
+##' - `port` (optional) -- The port for accessing the database. If
+##' omitted, this will use the PostgreSQL default (5432).
+##'
+##' @param params Named list of database connection options. See details
+##' @return Database connection object
 ##' @author Rob Kooper
 ##' @export
 ##' @examples
 ##' \dontrun{
 ##' db.open(settings$database$bety)
 ##' }
 db.open <- function(params) {
-  params$dbfiles <- NULL
-  params$write <- NULL
+  if ("dbfiles" %in% names(params)) {
+    PEcAn.logger::logger.warn(
+      "`dbfiles` in the database driver is deprecated. Setting it to NULL."
+    )
+    params[["dbfiles"]] <- NULL
+  }
+  if ("write" %in% names(params)) {
+    PEcAn.logger::logger.warn(
+      "`write` in the database driver is deprecated. Setting it to NULL."
+    )
+    params[["write"]] <- NULL
+  }
 
-  if(is.null(params$driver) || params$driver == "PostgreSQL") {
-    requireNamespace("RPostgreSQL")
+  driver <- params[["driver"]]
+  params[["driver"]] <- NULL
+  if (is.null(driver)) {
+    PEcAn.logger::logger.info(
+      "Missing `driver` argument. ",
+      "Assuming `RPostgreSQL::PostgreSQL()` driver."
+    )
+    driver <- "PostgreSQL"
   }
 
-  if (is.null(params$driver)) {
-    args <- c(drv = DBI::dbDriver("PostgreSQL"), params, recursive = TRUE)
-  } else {
-    args <- c(drv = DBI::dbDriver(params$driver), params, recursive = TRUE)
-    args[['driver']] <- NULL
+  if (!driver %in% c("PostgreSQL", "Postgres")) {
+    PEcAn.logger::logger.severe(paste0(
+      "Driver `", driver, "` is not supported. ",
+      'You must use either `"PostgreSQL"` or `"Postrgres"`.'
```

Comment:
Is `postrgres` a typo or a driver I haven't seen before?

Reply:
Good catch! I was _this_ close to sending you a link about the `RPostgres` package before realizing that this was, indeed, a typo.

Reply:
Fixed in 993b387dc0214ece8ee2c343793e4b0220156bec.

Reply:
It totally scans like an R package name. `postrGREs` is the data package a presentation about standardized tests, `postRgres` is for post-hoc calculations on continuous linear models, and `postrGres` is the R frontend for a database of who's most active in your Tumblr comments.